index.js

```diff
@@ -198,6 +204,32 @@ Babel.prototype.processString = function(string, relativePath) {
   var transpiled = this.transform(string, options);
   var key = options.moduleId ? options.moduleId : relativePath;
 
+  if (this.helperWhiteList) {
+    var invalidHelpers = transpiled.metadata.usedHelpers.filter(function(helper) {
+      return this.helperWhiteList.indexOf(helper) === -1;
+    }, this);
+
+    if (invalidHelpers.length > 0) {
+      var message = relativePath + ' was transformed and relies on `' + invalidHelpers[0] + '`, which was not included in the helper whitelist. Either add this helper to the whitelist or refactor to not be dependent on this runtime helper.';
+
+      if (invalidHelpers.length > 1) {
+        var helpers = invalidHelpers.map(function(item, i) {
+          if (i === invalidHelpers.length - 1) {
+            return '& `' + item;
+          } else if (i === invalidHelpers.length - 2) {
+            return item + '`, ';
+          }
+
+          return item + '`, `';
+        }).join('');
+
+        message = relativePath + ' was transformed and relies on `' + helpers + '`, which were not included in the helper whitelist. Either add these helpers to the whitelist or refactor to not be dependent on these runtime helper.';
+      }
+
+      throw new Error(message);
+    }
+  }
```

Comment:
would it make sense to extract that into a separate method/function to reduce the code complexity?